toolkit/themes/shared/design-system/config.js

```diff
@@ -82,64 +100,150 @@ ${formattedVars}
 }
 
 /**
- * Takes a token object and changes "value" based on the supplied prop. Also
- * preserves variable references when necessary.
+ * Finds the original value of a token for a given media query and surface.
+ * 
+ * @param {object} token - Token object parsed by style-dictionary.
+ * @param {string} prop - Name of the property we're querying for. 
+ * @param {string} surface 
+ *  The desktop surface we're generating CSS for, either "brand" or "platform".
+ * @returns {string} The original token value based on our parameters.
+ */
+function getOriginalTokenValue(token, prop, surface) {
+  if (surface) {
+    return token.original.value[surface]?.[prop];
+  } else if (prop == "default" && typeof token.original.value != "object") {
+    return token.original.value;
+  }
+  return token.original.value?.[prop];
+}
+
+/**
+ * Updates a token's value to the relevant original value after resolving
+ * variable references.
  *
  * @param {object} token - Token object parsed from JSON by style-dictionary.
- * @param {string} prop
- *  Name of the property used to get the token's new value.
- * @param {object} dictionary
+ * @param {string} originalVal
+ *  Original value of the token for the combination of surface and media query.
+ * @param {object} dictionary 
  *  Object of transformed tokens and helper fns provided by style-dictionary.
  * @returns {object} Token object with an updated value.
  */
-function transformTokenValue(token, prop, dictionary) {
-  let originalVal = token.original.value[prop];
-  if (dictionary.usesReference(originalVal)) {
-    let refs = dictionary.getReferences(originalVal);
-    return { ...token, value: `var(--${refs[0].name})` };
+function transformTokenValue(token, originalVal, dictionary) {
+  let value = originalVal;
+  if (dictionary.usesReference(value)) {
+    dictionary.getReferences(value).forEach(ref => {
+      value = value.replace(`{${ref.path.join(".")}}`, `var(--${ref.name})`);
+    });
   }
-  return { ...token, value: token[prop] || originalVal};
+  return { ...token, value };
+}
+
+/**
+ * Creates a light-dark transformed that works for a given surface. Registers
```

Comment:
```suggestion
 * Creates a light-dark transformer that works for a given surface. Registers
```
Think there's a typo here or a missing word, not sure which it is 😅